### PR TITLE
Feature/disable backspace reset

### LINKED ISSUE #926: Prevent backspace key from clearing selected value
Is there a way to prevent backspace key from clearing the selected value in ui-select?

Thank you in advance

### PR DESCRIPTION
**Pull request by pierregaste**
Add a new option that prevent the backspace key reset the model (set to undefined).
Th option could be added to the directive ```ui-select```:
```html
<ui-select disable-backspace-reset="true"></ui-select>
```

I hope I correctly commit and implement this feature! :smiley: 
***
feat(uiSelectSingleDirective): add an option to avoid backspace model reset

Currently, when you select an option and you press the backspace key, it will reset the model.
With this option, we could disable this behavior to avoid the model resetting.

Closes #926 #525

### CODE REVIEW

**Comment by deeg** [comment]
I'm not sure how the other maintainers feel about adding this feature, but before it can be merged the PR will need to be squashed and rebased.

I will let @aaronroberson and @user378230 weigh in on if they think this is a feature they would want to merge.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,13 @@
+<a name="0.14.10"></a>
```
Comment (deeg) [inline CHANGELOG.md]: I believe this will get created automatically at build time once we fix everything up.

Please remove these changes from the file.

**Comment by deeg** [comment]
Also please remove the /dist files from the PR.

**Comment by pierregaste** [comment]
I just rebased the PR. Is it OK for you?

**Comment by user378230** [comment]
Does this have any unit tests?

What issue is this actually trying to solve? 

Can you provide:
- a plunkr based on current library and the problem (along with steps to follow) 
- a plunkr with your changes and describe what difference this will make to the user

Thanks for your contribution but I can't merge and maintain changes I don't know anything about.

**Comment by user378230** [comment]
- This still needs a test 
- Squashing to a single commit. 
- Can you rebase and add an example? It should be fairly easy to do. _Examples have been moved to docs/examples now._ 
- Can post here some documentation here, which can be included in wiki when the PR is merged.

**Comment by pierregaste** [comment]
I removed this fork by mistake.
Please go to https://github.com/angular-ui/ui-select/pull/1736.

**Pull request closed without merging**